File: kyopro-100-py/1_itp1_7_b.py
while True:
    l = list(map(int, input().split()))
    n,x = l[0], l[1]
    if n==0 and x==0:
        break
    else:
        cnt = 0
        for i in range(1,n-1):
            for j in range(i+1, n):
                for k in range(j+1, n+1):
                    if (i+j+k) == x:
                        cnt += 1
        print(cnt)

File: src/bin/1_itp1_7_b.rs
use proconio::input;

fn main() {
    let mut flag = true;
    while flag {
        input! {
            l: [i32; 2],
        };
        let mut cnt = 0;
        let n = l[0];
        let x = l[1];
        if n == 0 && x == 0 {
            flag = false;
        } else {
            for i in 1..(n-1) {
                for j in (i+1)..n {
                    for k in (j+1)..(n+1) {
                        if (i+j+k) == x {
                            cnt += 1
                        }
                    }
                }
            }
        println!("{}", cnt);
        }
    }
}